src/java/org/apache/cassandra/db/streaming/CassandraStreamHeader.java

```diff
@@ -65,18 +85,43 @@ private CassandraStreamHeader(Version version, SSTableFormat.Type format, long e
         this.compressionInfo = compressionInfo;
         this.sstableLevel = sstableLevel;
         this.header = header;
-
+        this.fullStream = fullStream;
+        this.componentManifest = componentManifest;
+        this.firstKey = firstKey;
+        this.tableId = tableId;
         this.size = calculateSize();
     }
 
-    public CassandraStreamHeader(Version version, SSTableFormat.Type format, long estimatedKeys, List<SSTableReader.PartitionPositionBounds> sections, CompressionMetadata compressionMetadata, int sstableLevel, SerializationHeader.Component header)
+    private CassandraStreamHeader(Version version, SSTableFormat.Type format, long estimatedKeys,
```

Comment:
The introduction of the new fields and constructors got us to 5 constructors total with up to 10 arguments, which is no longer manageable, and calls for a builder. It's boring and tedious work, so I did it myself and pushed here - 
https://github.com/iamaleksey/cassandra/commit/321d21747faa46afcf34518ebdeb811f2a805de8 - please feel free to cherry-pick.

In addition to introducing the builder, the commit renames `fullStream` to something a bit more meaningful (`isEntireSSTable`) that clearly reflects what's actually happening, fixes a bug in `serializedSize()` where compression info isn't initialized, and removes some fields without `toString()` implementations from header's own `toString()`.

Reply:
I have cherry picked this change. Thanks!